src/mpid/ch4/shm/posix/eager/fbox/fbox_types.h

```diff
@@ -14,20 +14,18 @@
 #include <mpidimpl.h>
 #include "mpidu_init_shm.h"
 
-#define MPIDI_POSIX_FBOX_DATA_LEN  (16 * 1024 - sizeof(uint64_t) - 2 * sizeof(int))
-#define MPIDI_POSIX_FBOX_THRESHOLD (MPIDI_POSIX_FBOX_DATA_LEN)
+#define MPIDI_POSIX_FBOX_SIZE       (16 * 1024)
 
 typedef struct {
-
-    volatile uint64_t data_ready;
-
+    MPL_atomic_int_t data_ready;
     int is_header;
     size_t payload_sz;
-
-    uint8_t payload[MPIDI_POSIX_FBOX_DATA_LEN];
-
+    uint8_t payload[];
 } MPIDI_POSIX_fastbox_t;
 
+#define MPIDI_POSIX_FBOX_DATA_LEN  (MPIDI_POSIX_FBOX_SIZE - sizeof(MPIDI_POSIX_fastbox_t))
```

Comment:
You cannot do `sizeof` on structs that have flexible arrays.

Reply:
Yes, you can. I believe that is the signature feature. According to wikipedia https://en.wikipedia.org/wiki/Flexible_array_member:

> The sizeof operator on such a struct gives the size of the structure as if the flexible array member was empty.

Reply:
Interesting.  I didn't know that.